src/shogun/classifier/NearestCentroid.cpp

```diff
@@ -72,18 +72,16 @@ namespace shogun{
 			num_per_class[current_class]++;
 		}
 
+		SGVector<float64_t> scale(num_classes);
 		for (int32_t i=0 ; i<num_classes ; i++)
 		{
-			auto target = centroids.get_column(i);
 			int32_t total = num_per_class[i];
-			float64_t scale = 0;
 			if(total>1)
-				scale = 1.0/((float64_t)(total-1));
+				scale[i] = 1.0/((float64_t)(total-1));
 			else
-				scale = 1.0/(float64_t)total;
-
-			target.scale(scale);
+				scale[i] = 1.0/(float64_t)total;
```

Comment:
no need to loop this if all values are the same. Just use some kind of set_const call ...

Reply:
the values are not the same. `total = num_per_class[i]`

Reply:
sorry I should have read that better :D